mwtext/utilities/preprocess_wikidata.py

```diff
@@ -0,0 +1,86 @@
+r"""
+``$ mwtext preprocess_wikidata -h``
+::
+    Converts WIkidata JSON dumps to string of property/value pairs.
+    Usage:
+        preprocess_wikidata (-h|--help)
+        preprocess_wikidata [<input-file>...]
+                            [--threads=<num>] [--output=<path>]
+                            [--compress=<type>] [--verbose] [--debug]
+    Options:
+        -h|--help           Print this documentation
+        <input-file>        The path to a Wikidata JSON Dump file
+                            [default: <stdin>]
+        --threads=<num>     If a collection of files are provided, how many
+                            processor threads? [default: <cpu_count>]
+        --output=<path>     Write output to a directory with one output file
+                            per input path.  [default: <stdout>]
+        --compress=<type>   If set, output written to the output-dir will be
+                            compressed in this format. [default: bz2]
+        --verbose           Print progress information to stderr.  Kind of a
+                            mess when running multi-threaded.
+        --debug             Print debug logs.
+"""
+
+import bz2
+import json
+import os
+import random
+import sys
+
+import mwbase
+import mwcli
+
+from ..wikidata_preprocessor import WikidataPreprocessor
+
+
+def preprocess_wikidata(dump, verbose=False):
+    wikidata_preprocessor = WikidataPreprocessor()
+    for item_json in dump:
```

Comment:
We'll be processing an XML dump here, so the pattern will be something like this: 

```python
for page in dump:
    for revision in page:
        if not is_relevant_page(page, revision):
            continue
        item_doc = json.loads(revision.text)
        entity = mwbase.Entity.from_json(item_doc)
        if not is_relevant_entity(entity):
            continue
        
        claims_tuples = list(wikidata_preprocessor.process(entity))
        random_shuffle(claims_tuples)
        ...
```

Reply:
what would make a page relevant in is_relevant_page?

Reply:
In our case, it might just be that it's namespace zero (Wikidata items). 

I'm not sure what else we might filter on. @geohci, any thoughts?

Reply:
@halfak two things:
* I had been assuming we'd just use the JSON dumps but I see now that that would either require extending mwxml or mwbase to handle the JSON dumps (not hard but not really what those libraries were intended for) or require a good bit of custom code for Wikidata to process the JSON dump that would make maintenance more of a hassle and break some of the consistency with the wikitext utilities. For the record though, I've never processed the wikidata XML dumps so I'm not sure if that introduces any new issues.
* Given that we're working with the XML dumps (I assume wikidatawiki-YYYYMMDD-pages-articles.xml.bz2 ?), namespace 0 is a good filter. the other would be that revision.model == 'wikibase-item' (filter out lexemes and any other non-QIDs -- doing a regex on the page title for Q#### should accomplish the same thing). those are the only two i can think of right now though there might be other pages in wikidata I'm not aware of that we'd want to skip.